cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -108,3 +112,46 @@ def transcript_credentials_handler(request, course_key_string):
             response = JsonResponse({'error': error_message}, status=400)
 
     return response
+
+
+@expect_json
+@login_required
+@require_GET
+def transcript_download_handler(request, course_key_string):
+    """
+    JSON view handler to download a transcript.
+
+    Arguments:
+        request: WSGI request object
+        course_key_string: course key
+        filename: Name of the to be created transcript file
+
+    Returns:
+        - A 200 response with SRT transcript file attached.
+        - A 400 if there is a validation error.
+        - A 404 if there is no such transcript or feature flag is disabled.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not VideoTranscriptEnabledFlag.feature_enabled(course_key):
+        return HttpResponseNotFound()
+
+    edx_video_id = request.GET.get('edx_video_id')
+    if not edx_video_id:
+        return JsonResponse({'error': 'edx_video_id is required.'}, status=400)
+
+    language_code = request.GET.get('language_code')
+    if not language_code:
+        return JsonResponse({'error': 'language code is required.'}, status=400)
+
+    transcript = get_video_transcript_data(video_ids=[edx_video_id], language_code=language_code)
+    if transcript:
+        basename, __ = os.path.splitext(transcript['file_name'])
+        transcript_filename = '{base_name}.srt'.format(base_name=basename.encode('utf8'))
```

Comment:
why do we need to do this?

Reply:
file name may come from client title of Video, so may contain unicode characters